Description: Added check with the invalid crq sent by Sean Burford
 .
 This is 0964ed7b45ef593f1e118abb0e742e093a5f7987 converted to quilt format.
Author: Nikos Mavrogiannopoulos <nmav@redhat.com>
Origin: upstream
Forwarded: not-needed

--- gnutls28-3.3.8.orig/configure.ac
+++ gnutls28-3.3.8/configure.ac
@@ -330,6 +330,7 @@ AC_ARG_ENABLE(non-suiteb-curves,
 if [ test "$enable_non_suiteb" = "yes" ];then
   AC_DEFINE([ENABLE_NON_SUITEB_CURVES], 1, [Enable all curves])
 fi
+AM_CONDITIONAL(ENABLE_NON_SUITEB_CURVES, test "$enable_non_suiteb" = "yes")
 
 AM_CONDITIONAL(ENABLE_SELF_CHECKS, test "$enable_self_checks" = "yes")
 if [ test "$enable_self_checks" = "yes" ];then
--- gnutls28-3.3.8.orig/tests/cert-tests/Makefile.am
+++ gnutls28-3.3.8/tests/cert-tests/Makefile.am
@@ -27,12 +27,16 @@ EXTRA_DIST = ca-no-pathlen.pem no-ca-or-
 	template-overflow.tmpl template-overflow2.pem template-overflow2.tmpl \
 	template-date.tmpl template-date.pem template-dn-err.tmpl \
 	template-nc.tmpl template-nc.pem xmpp-othername.pem \
-	suppressions.valgrind
+	suppressions.valgrind csr-invalid.der
 
-dist_check_SCRIPTS = pathlen aki template-test pem-decoding dane
+dist_check_SCRIPTS = pathlen aki template-test pem-decoding dane crq
 
 TESTS = pathlen aki pem-decoding
 
+if ENABLE_NON_SUITEB_CURVES
+TESTS += crq
+endif
+
 if !WINDOWS
 TESTS += template-test
 endif
--- /dev/null
+++ gnutls28-3.3.8/tests/cert-tests/crq
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+# Copyright (C) 2014 Red Hat, Inc.
+#
+# Author: Nikos Mavrogiannopoulos
+#
+# This file is part of GnuTLS.
+#
+# GnuTLS is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at
+# your option) any later version.
+#
+# GnuTLS is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GnuTLS; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+#set -e
+
+srcdir=${srcdir:-.}
+CERTTOOL=${CERTTOOL:-../../src/certtool$EXEEXT}
+DIFF=${DIFF:-diff}
+if ! test -z "${VALGRIND}";then
+VALGRIND="${LIBTOOL:-libtool} --mode=execute ${VALGRIND} --error-exitcode=15"
+fi
+
+$VALGRIND $CERTTOOL --inder --crq-info --infile $srcdir/csr-invalid.der >/dev/null 2>&1
+rc=$?
+
+# We're done.
+if test "$rc" != "1"; then
+  echo "Invalid crq decoding failed"
+  exit $rc
+fi
+
+exit 0
